src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -622,7 +632,7 @@ void handleReset(ResetFrame frame, Flow.Subscriber<?> subscriber) {
                 }
                 closed = true;
             } finally {
-                stateLock.lock();
+                stateLock.unlock();
```

Comment:
Ouch! I am to blame for this one. It's a regression introduced by JDK-8308310. This one needs to be backported to JDK 21. Let me see if I should prepare a PR for that.

Reply:
https://github.com/openjdk/jdk/pull/16101